src/dcqc/filesystems/synapsefs.py

```diff
@@ -0,0 +1,640 @@
+                info, whatever the value of `namespaces` may be.
+
+        Returns:
+            ~fs.info.Info: resource information object.
+
+        Raises:
+            fs.errors.ResourceNotFound: If ``path`` does not exist.
+
+        For more information regarding resource information, see :ref:`info`.
+
+        """
+        self.validatepath(path)
+        entity = self._path_to_entity(path)
+
+        raw_info = dict()
+        namespaces = namespaces or ()
+        name = entity.name
+        is_dir = is_container(entity)
+        is_file = not is_dir
+        raw_info["basic"] = {
+            "name": name,
+            "is_dir": not is_file,  # Folder and projects are both file containers
+        }
+
+        if "details" in namespaces:
+            size = entity._file_handle.contentSize if is_file else 0
+            type_ = ResourceType.file if is_file else ResourceType.directory
+
+            raw_info["details"] = {
+                "accessed": None,
+                "created": datetime_to_epoch(iso_to_datetime(entity.createdOn)),
+                "metadata_changed": datetime_to_epoch(
+                    iso_to_datetime(entity.modifiedOn)
+                ),
+                "modified": datetime_to_epoch(iso_to_datetime(entity.modifiedOn)),
+                "size": size,
+                "type": type_,
+                "_write": [],
+            }
+
+        if "synapse" in namespaces:
+            creator_id = int(entity.createdBy)
+            creator = self.synapse.getUserProfile(creator_id)
+            creator_username = creator["userName"]
+
+            modifier_id = int(entity.modifiedBy)
+            modifier = self.synapse.getUserProfile(modifier_id)
+            modifier_username = modifier["userName"]
+
+            raw_info["synapse"] = {
+                "concrete_type": entity.concreteType,
+                "short_type": entity.concreteType.split(".")[-1],
+                "creator_id": creator_id,
+                "creator_username": creator_username,
+                "etag": entity.etag,
+                "id": entity.id,
+                "modifier_id": modifier_id,
+                "modifier_username": modifier_username,
+                "parent_id": entity.parentId,
+                "content_type": None,
+                "content_md5": None,
+                "version_label": None,
+                "version_number": None,
+                "_write": [],
+            }
+
+            if is_file:
+                file_info = {
+                    "content_type": entity._file_handle.contentType,
+                    "content_md5": entity._file_handle.contentMd5,
+                    "version_label": entity.versionLabel,
+                    "version_number": entity.versionNumber,
+                }
+                raw_info["synapse"].update(file_info)
+
+        if "annotations" in namespaces:
+            raw_info["annotations"] = entity.annotations
+
+        return Info(raw_info)
```

Comment:
what does `Info` here do?  Does it just take kwargs?

Reply:
You can explore the class [here](https://github.com/PyFilesystem/pyfilesystem2/blob/8ed9dc495d8ba2f83fbb2a1145d34d92e13644be/fs/info.py#L30), but it's essentially a wrapper around the `raw_info` dictionary with several convenience functions. You can review [these docs](https://docs.pyfilesystem.org/en/latest/info.html) for more details on to use the `Info` class.

Reply:
Thanks for linking the docs.  Makes sense - looks like there are some nice class methods you are able to use in `Info`